### ISSUE TITLE
How to Update multiple Atoms at the same time with only one Re-render

### ISSUE THREAD

**Issue by ziongh** [pmndrs/jotai#1144]
I know that the idea behind atoms is that it should be used atomically, so ideally you would never need to update two separate atoms as a single action. But let's imagine a situation where we have a filter that has many properties (filter by name, age, and so on):

``` js

const filterNameAtom= atom(null);
const filterAgeAtom= atom(40);

```

And in some Component we have a button to reset all the filtering options to the initial state. I Know that there is the **useResetAtom** hook, but for the sake of simplicity:

```js

const nameSetter = useSetAtom(filterNameAtom);
const ageSetter = useSetAtom(filterAgeAtom);

const refreshAllFilters = () => {
    nameSetter(null);
    ageSetter(50);
}

```

In this case if there is a component that uses both of this atoms, this component would be rerendered twice. Is there any other way besides splitting the component that depend on both atoms so that each sub-component depends only on one atom?

Thanks

**Comment by AjaxSolutions** [comment]
I wondered myself how to do this.  MobX has the transaction feature.
https://mobx.js.org/api.html#transaction

**Comment by dai-shi** [comment]
It should not render twice, because React should batch them.
If it does, a minimal reproduction would be appreciated.

**Comment by TwistedMinda** [comment]
Found an [interesting discussion](https://stackoverflow.com/questions/53048495/does-react-batch-state-update-functions-when-using-hooks) about that.

**TL;DR**: 
- Before react 18
  - If setters are called inside a `Promise`, `setTimeout` or other "non-react events": will **not** batch updates
  - If setters are called inside "react-events": will batch updpates 
 
- After react 18: should batch even in "non-react events" like `Promise` or `setTimeout`

My question would be : is an async Promise in a write-only atom considered a `react-event` or a `Promise` ?
I would say it's a promise obviously, but with the introduction of Suspense i'm not sure.

I think we could close this and the OP can create a new issue when they have a reproducible example of un-expected behaviour

**Comment by dai-shi** [comment]
atom's `write` function is considered outside react lifecycle. So, the answer to your question is a `Promise`. Suspense is only related with `read` function.

Yeah, let's close. We could reopen upon request.
